### PR TITLE
removed controller tests, fixed broken home tests, added new test

### PR DESCRIPTION
**Pull request by pandey-shubh**
## Pull request type

Please check the type of change your PR introduces:

-  Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

- home page e2e tests with controller
- broken tests

Issue Number: N/A

## What is the new behavior?

- removed controller based home page tests
- fixed broken tests related to tokens on home page
- removed NFT related test from home page
- added new test related to copying pub key from home page

### CODE REVIEW

File: tests/utils.ts [hunk 1]
```diff
@@ -31,11 +35,8 @@ export async function getInnerText(page: Page, selector: string): Promise<string
 }
 
 export async function getControllerState(page: Page) {
```
Comment (ArpitAgrawal321) [inline tests/utils.ts:37]: Lets remove this fn

File: tests/utils.ts [hunk 2]
```diff
@@ -90,9 +91,9 @@ export async function changeLanguage(page: Page, language: "english" | "german"
   await page.click(`nav ul[role='listbox'] div >> text=${languageLabels[language]}`);
   // wait for controllerModule language to update
   await page.waitForFunction(
-    async (locale) => {
-      const state = window.sessionStorage.getItem("controllerModule") || window.localStorage.getItem("controllerModule");
-      const controllerModule = await JSON.parse(state as string)?.controllerModule;
+    (locale) => {
```
Comment (ArpitAgrawal321) [inline tests/utils.ts:94]: Remove controller module dependency from here as well.

File: tests/home.utils.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { Page } from "@playwright/test";
+import * as dotenv from "dotenv";
+
+import { ensureTextualElementExists, wait } from "./utils";
+
+dotenv.config({ path: ".env.testing" });
+
+export async function clickTokenIfAvailable(page: Page) {
+  const token_locator = page.locator("//div/h2[contains(text(),'Tokens')]/..//span/p");
```
Comment (ArpitAgrawal321) [inline tests/home.utils.ts:9]: Is using xPath reliable? If DOM structure changes we'll have to update this.
Better to use selector based on text `Tokens` only.

File: tests/home.utils.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    await ensureTextualElementExists(page, "Transfer Details");
+  }
+}
+
+export async function clickPubKeyIcon(page: Page) {
+  await page.click("//img[@alt='wallet icon']");
+}
+
+export async function ensureCopiedToastDisplayed(page: Page) {
+  await page.locator("//div[@class='text-center justify-self-center']").waitFor();
```
Comment (ArpitAgrawal321) [inline tests/home.utils.ts:22]: Same as before, xPath.

File: tests/home.utils.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  const token_locator = page.locator("//div/h2[contains(text(),'Tokens')]/..//span/p");
+  if ((await token_locator.isVisible()) === true) {
+    await token_locator.click();
+    await wait(1000);
+    await ensureTextualElementExists(page, "Transfer Details");
+  }
+}
+
+export async function clickPubKeyIcon(page: Page) {
+  await page.click("//img[@alt='wallet icon']");
```
Comment (ArpitAgrawal321) [inline tests/home.utils.ts:18]: Same as before, xPath.

File: tests/home.utils.ts (hunk 4 above)
Comment (pandey-shubh) [reply tests/home.utils.ts:22]: changed based on Text diaplayed

File: tests/utils.ts (hunk 2 above)
Comment (pandey-shubh) [reply tests/utils.ts:94]: removed

File: tests/utils.ts (hunk 1 above)
Comment (pandey-shubh) [reply tests/utils.ts:37]: removed getControllerState function

File: tests/home.utils.ts (hunk 5 above)
Comment (pandey-shubh) [reply tests/home.utils.ts:18]: replaced xpath with text based search

File: tests/home.utils.ts (hunk 3 above)
Comment (pandey-shubh) [reply tests/home.utils.ts:9]: replaced xpath with text based selector

**Review by ArpitAgrawal321 — APPROVED** [review]
lgtm

**Pull request merged**